cupy/core/raw.pyx

```diff
@@ -51,6 +51,15 @@ cdef class RawKernel:
         # only used when RawKernels are produced by a cubin/ptx RawModule
         self.file_path = None
 
+        # per-device, per-instance cache
+        try:
+            self._kernel_cache = [None] * runtime.getDeviceCount()
```

Comment:
This line forces an early cuda initialization.
Maybe its better to do a lazy start and do this in the `__call__`

Reply:
This is what causes chainer tests to fail

Reply:
I see. Fixed in 9336c82. 

Just curious, why didn't Chainer want an early context initialization?

Reply:
Its because of using multiprocessing in fork mode, its not allowed by cuda, so if the cuda initialization happens at import time then it breaks.